### PR TITLE
feat(clang-tidy, colcon-build, colcon-test): add include-eol-distros

### PR DESCRIPTION
**Pull request by h-ohta**
## Description

- add argument `include-eol-distros`

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: colcon-build/action.yaml [hunk 1]
```diff
@@ -19,6 +19,10 @@ inputs:
     description: ""
     required: false
     default: ${{ github.token }}
+  rosdep-update-arguments:
+    description: ""
+    required: false
+    default: ""
```
Comment (kenji-miyake) [inline colcon-build/action.yaml:25]: If there is no other candidate `rosdep` args, renaming the input would be better.

```suggestion
  include-eol-distros:
    description: ""
    required: false
    default: false
```

File: colcon-build/action.yaml [hunk 2]
```diff
@@ -51,7 +55,7 @@ runs:
       run: |
         package_paths=$(colcon list -p --packages-above-and-dependencies ${{ inputs.target-packages }} --base-paths . dependency_ws)
         sudo apt-get -yqq update
-        rosdep update
+        rosdep update ${{ inputs.rosdep-update-arguments }}
```
Comment (kenji-miyake) [inline colcon-build/action.yaml:58]: If you go https://github.com/autowarefoundation/autoware-github-actions/pull/207/files#r1119493824, for example, this part will be as follows:

```suggestion
        if [ "${{ inputs.include-eol-distros }}" = "true" ]; then
            rosdep update --include-eol-distros
        else
            rosdep update
        fi
```

**Comment by h-ohta** [comment]
@kenji-miyake Thank you for your proposal.
I'll check whether it works.

**Comment by h-ohta** [comment]
@kenji-miyake I confirmed this PS works in https://github.com/tier4/autoware.universe/actions/runs/4289329599/jobs/7472256854

File: clang-tidy/action.yaml [hunk 3]
```diff
@@ -25,6 +25,10 @@ inputs:
     description: ""
     required: false
     default: ${{ github.token }}
+  include-eol-distros:
+    description: ""
+    required: false
+    default: "false"
```
Comment (kenji-miyake) [inline clang-tidy/action.yaml:31]: By the way, this default value is not mandatory because we check if it's `"true"` or not.

```suggestion
```

Comment (h-ohta) [reply clang-tidy/action.yaml:31]: I agree with it. I'll delete

File: colcon-build/action.yaml [hunk 4]
```diff
@@ -19,6 +19,10 @@ inputs:
     description: ""
     required: false
     default: ${{ github.token }}
+  include-eol-distros:
+    description: ""
+    required: false
+    default: "false"
```
Comment (h-ohta) [inline colcon-build/action.yaml:25]:
```suggestion
```

File: colcon-test/action.yaml [hunk 5]
```diff
@@ -19,6 +19,10 @@ inputs:
     description: ""
     required: false
     default: ${{ github.token }}
+  include-eol-distros:
+    description: ""
+    required: false
+    default: "false"
```
Comment (h-ohta) [inline colcon-test/action.yaml:25]:
```suggestion
```

**Review by kenji-miyake — APPROVED** [review]

**Pull request merged**